docs/SETTINGS.md

```diff
@@ -0,0 +1,195 @@
+# Repository Settings
+
+---
+
+These are the default settings that should be configured in each repo, changes can be made as needed.
+The settings described here should serve as a baseline, it is more important that there is some setting in place rather 
+than the exact settings being used across every repo.
+
+After a repo's settings are configured this file can be safely deleted from the repo since it is not needed for the
+project to run.
+
+## General
+
+---
+
+### Default Branch
+- Default branch should be `main`
+
```

Comment:
We might want to remove the redundant `---` on H1 and H2 headers. The final rich text display looks like this:

![image](https://github.com/strangelove-ventures/oss-repo-template-golang/assets/29391744/d68a3255-6f1e-4821-9e03-66e3d8bd0daf)

The double underline is a little redundant in my opinion.